bndtools.api/src/org/bndtools/api/central/ICentral.java

```diff
@@ -0,0 +1,22 @@
+package org.bndtools.api.central;
+
+import org.bndtools.api.ModelListener;
+import org.eclipse.jdt.core.IJavaProject;
+import org.osgi.annotation.versioning.ProviderType;
+
+import aQute.bnd.build.Project;
+
+@ProviderType
+public interface ICentral {
+
+	Project getModel(IJavaProject project);
+
+	void changed(Project model);
+
+	void addModelListener(ModelListener m);
```

Comment:
Maybe model listener can better use the whiteboardpattern instead?

Reply:
I agree 100%.

This was more about me trying to limit the scope of my PR which had already grown into a behemoth. This part of the code is not particularly well tested in CI (which is to say, not really tested at all!) and so I had a fear that something might break. So the goal was to make the minimal changes I needed in this PR to get the automated testing working reliably, and also so that it could get merged so that it could be subjected to some real-world testing by us on the bleeding edge. Then I could then proceed with additional refactorings (like this) with a little bit more confidence that I haven't broken anything in a subtle or fundamental way. I can foresee this interface evolving significantly as we start to leverage the power of a more component-driven approach.

Reply:
It dosen't hurt to have those though...